app/controllers/admin/check_outs_controller.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module Admin
+  class CheckOutsController < ApplicationController
+    load_and_authorize_resource instance_name: :checkpoint, class: Checkpoint, except: [:index]
+    load_and_authorize_resource :checkpoint_unit, only: [:redirect_to_check]
+
+    def index; end
+
+    def show; end
+
+    def checkpoint_unit_autocomplete
+      checkpoint_units = CheckpointUnit.joins(:unit).where(checkpoint_id: @checkpoint.id)
```

Comment:
To show also dependent ones, maybe something like

```suggestion
      checkpoint_units = CheckpointUnit.joins(:checkpoint).where(checkpoint: { depends_on_checkpoint: @checkpoint }).or(CheckpointUnit.where(checkpoint_id: @checkpoint.id)
```